src/tree/JobTreeItem.ts

```diff
@@ -0,0 +1,101 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as moment from 'moment';
+import { IncomingMessage } from 'ms-rest';
+import { gitHubWebResource } from 'vscode-azureappservice/out/src/github/connectToGitHub';
+import { requestUtils } from 'vscode-azureappservice/out/src/utils/requestUtils';
+import { AzExtTreeItem, AzureParentTreeItem, IActionContext, TreeItemIconPath } from "vscode-azureextensionui";
+import { Conclusion, Status } from '../constants';
+import { convertConclusionToVerb, convertStatusToVerb, createGitHubRequestOptions, getGitHubAccessToken } from '../utils/gitHubUtils';
+import { getTimeElapsedString } from '../utils/timeUtils';
+import { treeUtils } from "../utils/treeUtils";
+import { ActionTreeItem } from './ActionTreeItem';
+import { IAzureResourceTreeItem } from './IAzureResourceTreeItem';
+import { GitHubStep, StepTreeItem } from './StepTreeItem';
+
+export type GitHubJob = {
+    id: number;
+    run_id: number;
+    run_url: string;
+    node_id: string;
+    head_sha: string;
+    url: string;
+    html_url: string;
+    status: Status;
+    conclusion: Conclusion;
+    started_at: Date;
+    completed_at: Date;
+    name: string;
+    steps: GitHubStep[];
+    check_run_url: string;
+};
+
+export class JobTreeItem extends AzureParentTreeItem implements IAzureResourceTreeItem {
+
+    public static contextValue: string = 'azureStaticJob';
+    public readonly contextValue: string = JobTreeItem.contextValue;
+    public parent: ActionTreeItem;
+    public data: GitHubJob;
+
+    constructor(parent: ActionTreeItem, data: GitHubJob) {
+        super(parent);
+        this.data = data;
+    }
+
+    public get iconPath(): TreeItemIconPath {
+        return treeUtils.getActionIconPath(this.data.status, this.data.conclusion);
+    }
+
+    public get id(): string {
+        return `${this.parent.id}/${this.data.id}`;
```

Comment:
I think that you don't need the parent.id.  Each id should be unique and `fullId` will contain the entire parent.